st2common/st2common/hooks.py

```diff
@@ -94,18 +96,34 @@ def _abort_other_errors():
         }), status=500)
 
     @staticmethod
-    def _validate_token(headers):
-        """Validate token"""
-        if 'X_Auth_Token' not in headers:
-            LOG.audit('Token is not found in header.')
+    def _validate_token(request):
```

Comment:
Yeah, duplicated code in auth middleware and hook also makes me unhappy. We should refactor it.